internal/config/global_config_test.go

```diff
@@ -0,0 +1,103 @@
+package config_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/commitdev/zero/internal/config"
+	"github.com/stretchr/testify/assert"
+)
+
+const baseTestFixturesDir = "../../tests/test_data/configs/"
+
+var testCredentialFile = func() (func() string, func()) {
+	tmpConfigPath := getTmpConfig()
+	mockFunc := func() string { return tmpConfigPath }
+	teardownFunc := func() { os.RemoveAll(tmpConfigPath) }
+	return mockFunc, teardownFunc
+}
+
+func getTmpConfig() string {
+	pathFrom := path.Join(baseTestFixturesDir, fmt.Sprintf("credentials%s.yml", ""))
+	pathTo := path.Join(baseTestFixturesDir, fmt.Sprintf("credentials%s.yml", "-tmp"))
+	copyFile(pathFrom, pathTo)
+	return pathTo
+}
+
+func copyFile(from string, to string) {
+	bytesRead, err := ioutil.ReadFile(from)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = ioutil.WriteFile(to, bytesRead, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+func TestReadOrCreateUserCredentialsFile(t *testing.T) {
```

Comment:
These should really be testing the API of the package rather than the internals. By calling a function like `ReadOrCreateUserCredentialsFile` it means that function now has to be exported, and I would think the only exported functions would be `LoadUserCredentials` and `Save`

Reply:
Make sense, should I create another file for tests like this under the `config` package? so I can keep them private and still test them?

Reply:
Maybe, though the general practice is try to make your tests in the _test package, then test the exported stuff and not the internals so that your test is coupled to the "API" which should be stable and not the internals which could change. And if you get to the point where the "API" is masking a massive amount of code, try to split it up into smaller packages which can be easily tested in isolation.